site/src/modules/workspaces/WorkspaceTiming/Chart/utils.ts

```diff
@@ -0,0 +1,57 @@
+export type TimeRange = {
+	startedAt: Date;
+	endedAt: Date;
+};
+
+/**
+ * Combines multiple timings into a single timing that spans the entire duration
+ * of the input timings.
+ */
+export const mergeTimeRanges = (ranges: TimeRange[]): TimeRange => {
+	const sortedDurations = ranges
+		.slice()
+		.sort((a, b) => a.startedAt.getTime() - b.startedAt.getTime());
```

Comment:
I think this approach is easier to read. Do you have any concerns about it?

Reply:
For me the other way is more readable, because I think the sorting distracts what this is really trying to do (picking the smallest or biggest).  The sort really is only doing that as a side effect.

And, index access like `arr[arr.length - 1]` and `arr[0]` sketches me out.  For example, suppose `arr` is empty.  There may be some length check elsewhere, but it is not visible from this function.

Plus, it is less efficient.  But, I would not say these are major concerns in this case.